sdk/iot/hub/samples/src/paho_iot_hub_sas_telemetry_example.c

```diff
@@ -0,0 +1,418 @@
+  uint32_t sas_expiration = get_expiration_time(iot_hub_sas_key_expiration);
+
+  // Decode the base64 encoded SAS key to use for HMAC signing
+  az_span decoded_key_span;
+  if (az_failed(
+          res = sample_base64_decode(
+              iot_hub_sas_key_span, AZ_SPAN_FROM_BUFFER(sas_b64_decoded_key), &decoded_key_span)))
+  {
+    printf("Could not decode the SAS key: return code %d\n", res);
+    return res;
+  }
+
+  // Get the signature which will be signed with the decoded key
+  az_span sas_signature_span;
+  if (az_failed(
+          res = az_iot_hub_client_sas_get_signature(
+              &client,
+              sas_expiration,
+              AZ_SPAN_FROM_BUFFER(sas_signature_buf),
+              &sas_signature_span)))
+  {
+    printf("Could not get the signature for SAS key: return code %d\n", res);
+    return res;
+  }
+
+  // HMAC the signature with the decoded key
+  az_span encoded_span = AZ_SPAN_FROM_BUFFER(sas_signature_hmac_encoded_buf);
+  if (az_failed(
+          res
+          = sample_hmac_encrypt(decoded_key_span, sas_signature_span, encoded_span, &encoded_span)))
+  {
+    printf("Could not encrypt the signature: return code %d\n", res);
+    return res;
+  }
+
+  // base64 encode the result of the HMAC
+  az_span encoded_span_b64;
+  if(az_failed(res = sample_base64_encode(
+      encoded_span, AZ_SPAN_FROM_BUFFER(sas_signature_encoded_buf_b64), &encoded_span_b64)))
+  {
+    printf("Could not base64 encode the password: return code %d\n", res);
+    return res;
+  }
+
+  // Get the resulting password, passing the base64 encoded, HMAC signed bytes
+  size_t sas_key_length;
+  if (az_failed(
+          res = az_iot_hub_client_sas_get_password(
+              &client,
+              encoded_span_b64,
+              sas_expiration,
+              AZ_SPAN_NULL,
+              sas_key_password,
+              sizeof(sas_key_password),
+              &sas_key_length)))
+  {
+    printf("Could not get the password: return code %d\n", res);
+    return res;
+  }
+
+  return AZ_OK;
+}
+
+static int connect_device()
+{
+  int rc;
+
+  MQTTClient_SSLOptions mqtt_ssl_options = MQTTClient_SSLOptions_initializer;
+  MQTTClient_connectOptions mqtt_connect_options = MQTTClient_connectOptions_initializer;
+
+  // NOTE: We recommend setting clean session to false in order to receive any pending messages
+  mqtt_connect_options.cleansession = false;
+  mqtt_connect_options.keepAliveInterval = AZ_IOT_DEFAULT_MQTT_CONNECT_KEEPALIVE_SECONDS;
+
+  // Get the MQTT user name used to connect to IoT Hub
+  if (az_failed(
+          rc
+          = az_iot_hub_client_get_user_name(&client, mqtt_username, sizeof(mqtt_username), NULL)))
+
+  {
+    printf("Failed to get MQTT clientId, return code %d\n", rc);
+    return rc;
+  }
+
+  // This sample uses SAS authentication so the password field is set to the generated password
+  mqtt_connect_options.username = mqtt_username;
+  mqtt_connect_options.password = sas_key_password;
```

Comment:
Same comment for "sas_key_password". 
In this context it should be called "mqtt_password", as @CIPop has proposed.